auth.go

```diff
@@ -173,6 +186,10 @@ func NewTokenIntrospector(
 		opt(&options)
 	}
 
+	if options.userAgent == "" {
```

Comment:
It might be better to use a specified user agent as a prefix, similar to how [github.com/grpc/grpc-go](https://github.com/grpc/grpc-go) does. Currently, the user agent for gRPC requests is "go-authkit/v0.16.0 grpc-go/1.64.1".

With this approach, the final user agent would be:
- For gRPC: "<user_agent> go-authkit/v0.16.0 grpc-go/1.64.1"
- For HTTP: "<user_agent> go-authkit/v0.16.0"

@StiegOlSauer, what do you think?